crates/bevy_color/src/color_representation.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{Hsla, Lcha, LinearRgba, Oklaba, Srgba};
+
+/// An enumerated type that can represent any of the color types in this crate.
+///
+/// This is useful when you need to store a color in a data structure that can't be generic over
+/// the color type.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ColorRepresentation {
```

Comment:
I'd like to put some docs/constraints here on whether the alpha-colors are premultiplied or separate alpha.

Reply:
We should figure out what variant we _do_ use (assuming we only use one system), and document that these color types operate under that imo.